src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -2337,4 +2462,166 @@ RppStatus rppt_pixelate_gpu(RppPtr_t srcPtr,
 #endif // backend
 }
 
+/******************** fog ********************/
+
+RppStatus rppt_fog_gpu(RppPtr_t srcPtr,
+                       RpptDescPtr srcDescPtr,
+                       RppPtr_t dstPtr,
+                       RpptDescPtr dstDescPtr,
+                       Rpp32f *gammaTensor,
+                       RpptROIPtr roiTensorPtrSrc,
+                       RpptRoiType roiType,
+                       rppHandle_t rppHandle)
+{
+#ifdef HIP_COMPILE
+    RpptROI roiTensorPtrSrcHost[dstDescPtr->n];
+    CHECK_RETURN_STATUS(hipMemcpy(roiTensorPtrSrcHost, roiTensorPtrSrc, dstDescPtr->n * sizeof(RpptROI), hipMemcpyDeviceToHost));
+
+    RpptInterpolationType interpolationType = RpptInterpolationType::NEAREST_NEIGHBOR;
+    RppLayoutParams layoutParams = get_layout_params(srcDescPtr->layout, srcDescPtr->c);
+    Rpp32u height = srcDescPtr->h;
+    Rpp32u width = srcDescPtr->w;
+
+    RpptLayout layout = RpptLayout::NCHW;
+    RppLayoutParams srcLayoutParams = get_layout_params(layout, 1);
+    RpptDesc fogMaskSrcDesc ;
+    RpptDescPtr fogMaskSrcDescPtr = &fogMaskSrcDesc; // Fog original mask source description pointer
+    fogMaskSrcDescPtr->numDims = 3;
+    fogMaskSrcDescPtr->offsetInBytes = 0;
+    fogMaskSrcDescPtr->dataType = RpptDataType::F32;  
+    fogMaskSrcDescPtr->n = 1;
+    fogMaskSrcDescPtr->c = 1;
+    fogMaskSrcDescPtr->h = FOG_MAX_HEIGHT;
+    fogMaskSrcDescPtr->w = FOG_MAX_WIDTH; 
+    fogMaskSrcDescPtr->strides.nStride = FOG_MAX_WIDTH * FOG_MAX_HEIGHT;
+    fogMaskSrcDescPtr->strides.hStride = FOG_MAX_WIDTH;
+    fogMaskSrcDescPtr->strides.wStride = 1;
+    fogMaskSrcDescPtr->strides.cStride = 1;
+    fogMaskSrcDescPtr->layout = layout;
+
+    RpptDesc fogMaskDstDesc ;
+    RpptDescPtr fogMaskDstDescPtr = &fogMaskDstDesc; //Fog resized mask destination description pointer
+    fogMaskDstDescPtr->numDims = 3;
+    fogMaskDstDescPtr->offsetInBytes = 0;
+    fogMaskDstDescPtr->dataType = RpptDataType::F32;  
+    fogMaskDstDescPtr->n = 1;
+    fogMaskDstDescPtr->c = 1;
+    fogMaskDstDescPtr->h = height;
+    fogMaskDstDescPtr->w = width;
+    fogMaskDstDescPtr->strides.nStride = height * width;
+    fogMaskDstDescPtr->strides.hStride = width;
+    fogMaskDstDescPtr->strides.wStride = 1;
+    fogMaskDstDescPtr->strides.cStride = 1;
+    fogMaskDstDescPtr->layout = layout;
+
+    RpptImagePatchPtr internalDstImgSizes = reinterpret_cast<RpptImagePatch *>(rpp::deref(rppHandle).GetInitHandle()->mem.mgpu.scratchBufferPinned.floatmem);
+    RpptROI *internalRoiTensorPtrSrc = reinterpret_cast<RpptROI *>(internalDstImgSizes + 2);
+
+    internalDstImgSizes->width = width;
+    internalDstImgSizes->height = height;
+    internalRoiTensorPtrSrc->xywhROI.xy.x = 0 ;
```

Comment:
can we directly assign values in single line, like how it is done below?
https://github.com/ROCm/rpp/blob/develop/utilities/test_suite/HOST/Tensor_host.cpp#L1610

Same comment for internalDstImgSizes
we can try directly assigning as below?
internalDstImgSizes = {width, height}